command/agent/job_endpoint.go

```diff
@@ -118,6 +118,11 @@ func (s *HTTPServer) JobSpecificRequest(resp http.ResponseWriter, req *http.Requ
 	case strings.HasSuffix(path, "/action"):
 		jobID := strings.TrimSuffix(path, "/action")
 		return s.jobRunAction(resp, req, jobID)
+	case strings.HasSuffix(path, "/tag"):
+		parts := strings.Split(path, "/")
+		jobID := parts[0]
+		name := parts[2] // job/<jobID>/tag/<name>
```

Comment:
You need a length check here, otherwise `job/:jobid/tag` will panic rather than 404.